--- oathtool/gl/Makefile.orig	2013-06-08 21:21:02.211862052 -0700
+++ oathtool/gl/Makefile	2013-06-08 21:22:24.251603964 -0700
@@ -912,11 +912,11 @@
 # statements but through direct file reference. Therefore this snippet must be
 # present in all Makefile.am that need it. This is ensured by the applicability
 # 'all' defined above.
 BUILT_SOURCES = $(ALLOCA_H) $(ERRNO_H) $(FLOAT_H) parse-datetime.c \
 	arg-nonnull.h c++defs.h warn-on-use.h $(STDARG_H) $(STDBOOL_H) \
-	$(STDDEF_H) $(STDINT_H) stdio.h stdlib.h string.h sys/time.h \
+	$(STDDEF_H) $(STDINT_H) stdlib.h string.h sys/time.h \
 	time.h unistd.h wchar.h
 SUFFIXES = 
 MOSTLYCLEANFILES = core *.stackdump alloca.h alloca.h-t errno.h \
 	errno.h-t float.h float.h-t parse-datetime.c-t arg-nonnull.h \
 	arg-nonnull.h-t c++defs.h c++defs.h-t warn-on-use.h \
